zulipterminal/ui_tools/buttons.py

```diff
@@ -67,33 +70,17 @@ def update_count(self, count: int, text_color: Optional[str] = None) -> None:
     def update_widget(
         self, count_text: Tuple[Optional[str], str], text_color: Optional[str]
     ) -> Any:
-        # Note that we don't modify self._caption
-        max_caption_length = self.width_for_text_and_count - len(count_text[1])
-        if len(self._caption) > max_caption_length:
-            caption = (
-                self._caption[: max_caption_length - 1] + "\N{HORIZONTAL ELLIPSIS}"
-            )
+        if self.prefix_character:
+            prefix = [" ", self.prefix_character, " "]
         else:
-            caption = self._caption
-        num_extra_spaces = (
-            self.width_for_text_and_count - len(count_text[1]) - len(caption)
-        )
-
-        # NOTE: Generated text does not include space at end
-        # NOTE: Some are styles, so are not included in f-string
-        self._w = urwid.AttrMap(
-            urwid.SelectableIcon(
-                [
-                    " ",
-                    self.prefix_character,
-                    f"{self.post_prefix_spacing}{caption}{num_extra_spaces * ' '} ",
-                    count_text,
-                ],
-                self.width_for_text_and_count + 5,  # cursor location
-            ),
-            text_color,
-            "selected",
-        )
+            prefix = " "
+        if count_text[1]:
+            suffix = [" ", count_text, " "]
+        else:
+            suffix = " "
```

Comment:
But why do we need 2 spaces at the end?

Reply:
This reproduces the current behavior - I think otherwise it can look a little strange with the text between all of the unread counts, though I'm not sure if the current behavior was intentional. We can see what others think, though this won't be a pure refactor if we change the appearance.